core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/withEditor.ts

```diff
@@ -0,0 +1,306 @@
+                        mapContainerStructureToArrayOfContainers(pageAPIResponse.containers);
+                    const personalization = getPersonalization(pageAPIResponse.viewAs?.persona);
+
+                    return {
+                        containers,
+                        personalization,
+                        id: pageAPIResponse.page.identifier,
+                        languageId: pageAPIResponse.viewAs.language.id,
+                        personaTag: pageAPIResponse.viewAs.persona?.keyTag
+                    };
+                }),
+                $reloadEditorContent: computed<ReloadEditorContent>(() => {
+                    return {
+                        code: store.pageAPIResponse()?.page?.rendered,
+                        isTraditionalPage: store.isTraditionalPage(),
+                        isEditState: store.isEditState(),
+                        isEnterprise: store.isEnterprise()
+                    };
+                }),
+                $editorIsInDraggingState: computed<boolean>(
+                    () => store.state() === EDITOR_STATE.DRAGGING
+                ),
+                $editorProps: computed<EditorProps>(() => {
+                    const pageAPIResponse = store.pageAPIResponse();
+                    const socialMedia = store.socialMedia();
+                    const ogTags = store.ogTags();
+                    const device = store.device();
+                    const canEditPage = store.canEditPage();
+                    const isEnterprise = store.isEnterprise();
+                    const state = store.state();
+                    const params = store.params();
+                    const isTraditionalPage = store.isTraditionalPage();
+                    const contentletArea = store.contentletArea();
+                    const bounds = store.bounds();
+                    const dragItem = store.dragItem();
+                    const isEditState = store.isEditState();
+
+                    const isDragging = state === EDITOR_STATE.DRAGGING;
+                    const dragIsActive = isDragging || state === EDITOR_STATE.SCROLL_DRAG;
+
+                    const isLoading = store.status() === UVE_STATUS.LOADING;
+                    const isScrolling =
+                        state === EDITOR_STATE.SCROLL_DRAG || state === EDITOR_STATE.SCROLLING;
+
+                    const url = sanitizeURL(params.url);
+
+                    const pageAPIQueryParams = createPageApiUrlWithQueryParams(url, params);
+
+                    const showDialogs = canEditPage && isEditState;
+
+                    const showContentletTools =
+                        !!contentletArea && canEditPage && isEditState && !isScrolling;
+
+                    const showDropzone = canEditPage && isDragging;
+
+                    const showPalette = isEnterprise && canEditPage && isEditState;
+
+                    const shouldShowSeoResults = socialMedia && ogTags;
```

Comment:
All this might be a `signalStoreFeature` called `withEditorFlags`

Reply:
Not sure. Any thought is useful here @dotCMS/frontend

Reply:
I dont think this should be part of the store, can be util functions as arcadio said above